### PR TITLE
[Java] Archive logging improvements

### PR DESCRIPTION
**Pull request by vyazelenko**
This PR contains the following improvements:
- Add logging for `ReplicationSession` state change
- Add logging for `ControlSession` state change
- Add new state `ControlSession.State.CLOSING` and a new state transition from `INACTIVE` to `CLOSING`, i.e. this allows distinguishing from normal session termination (via `abort`) vs abnormal termination
- Add logging for errors in `ReplaySession`
- Make `Election` non-public and adjust logging interceptor
- Fix `ControlSession#hasNoActivity` check

### CODE REVIEW

File: aeron-agent/src/main/java/io/aeron/agent/ArchiveEventEncoder.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2019 Real Logic Ltd.
```
Comment (mjpt777) [inline aeron-agent/src/main/java/io/aeron/agent/ArchiveEventEncoder.java:2]: Copyright needs to be for range.

File: aeron-agent/src/test/java/io/aeron/agent/ArchiveEventEncoderTest.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        final int offset = 24;
+        final TimeUnit from = DAYS;
+        final TimeUnit to = MILLISECONDS;
+        final long sessionId = Long.MAX_VALUE;
+        final String payload = from.name() + STATE_SEPARATOR + to.name();
+        final int length = payload.length() + SIZE_OF_LONG + SIZE_OF_INT;
+        final int captureLength = captureLength(length);
+
+        final int encodedLength =
+            encodeSessionStateChange(buffer, offset, captureLength, length, from, to, sessionId);
```
Comment (mjpt777) [inline aeron-agent/src/test/java/io/aeron/agent/ArchiveEventEncoderTest.java:51]: only put args on next line.

File: aeron-archive/src/main/java/io/aeron/archive/ReplaySession.java [hunk 3]
```diff
@@ -253,12 +253,24 @@ Counter limitPosition()
...
+            onPendingError(sessionId, recordingId, errorMessage);
+            final ControlSession controlSession = this.controlSession;
+            if (!controlSession.isDone())
+            {
+                controlSession.attemptErrorResponse(correlationId, errorMessage, controlResponseProxy);
+            }
         }
     }
 
+    // Hook for Agent logging
```
Comment (mjpt777) [inline aeron-archive/src/main/java/io/aeron/archive/ReplaySession.java:268]: Put comment in body.

**Comment by mjpt777** [comment]
Looks good, only a few minor tweaks needed.

**Comment by vyazelenko** [comment]
@mjpt777 I've pushed fixes for the items you identified.

**Pull request merged**